.github/workflows/format.yml

```diff
@@ -0,0 +1,56 @@
+name: Format check
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**.py'
+
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+    outputs:
+      files: ${{ steps.git-diff-filter.outputs.files }}
+    name: Detect added and changed files
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: GrantBirki/git-diff-action@v2
+        id: git-diff-action
+        with:
+          base_branch: origin/main
+          search_path: '**.py'
+          # Ignore deleted files
+          git_options: '--no-color --diff-filter=d'
+
+      - name: Filter json diff
+        id: git-diff-filter
+        env:
+          JSON_DIFF: ${{ steps.git-diff-action.outputs.json-diff }}
+        run: |
+          files=$(echo $JSON_DIFF | jq -c -r '[.files[] | {path: .path}]')
+          echo "files=$files" >> $GITHUB_OUTPUT
+
+  check-files:
+    needs: generate
+    if: ${{ needs.generate.outputs.files != '[]' }}
+    runs-on: ubuntu-latest
+    # Allow the workflow run to pass when this job fails
+    continue-on-error: true
```

Comment:
Does this just change red to green? If yes then I don't like it, sorry for the confusion. 

https://github.com/orgs/community/discussions/15452

If formatting fails then the check should be red to draw attention, then maintainers should be able to merge anyway.

Reply:
No, it's still red, but won't block merging the PR.

Reply:
I'm still not sure `continue-on-error` is what you need here. Unfortunately, it's very confusing and barely documented - just one cryptic line https://github.com/actions/runner/issues/2347

Are you looking for `fail-fast: false` instead by any chance? Also: much better documented:

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

Reply:
Whatever happens with `continue-on-error`, from past experience with `fail-fast: false` I'm 99% sure you need it here.

Reply:
`continue-on-error` during my testing showed that jobs failed and showed up as red. But the workflow was still successful.
I've added the `fail-fast: false`.

Reply:
I'm still not sure about `continue-on-error` but this can easily be changed later. Also, I know first hand it's incredibly hard to test Github Actions before they are merged.

Reply:
Tested this on PR #768, seems to be working.